colcon_mixin/mixin/repository.py

```diff
@@ -16,8 +18,30 @@
 
 logger = colcon_logger.getChild(__name__)
 
-"""The path of the yaml file describing the mixin repositories."""
-mixin_repositories_file = get_config_path() / 'mixin_repositories.yaml'
+
+def get_mixin_repositories_file():
+    """
+    Get the path of the yaml file describing the mixin repositories.
+
+    :rtype: Path
+    """
+    return get_config_path() / 'mixin_repositories.yaml'
+
+
+if sys.version_info[:2] >= (3, 7):
+    def __getattr__(name):
+        if name == 'mixin_repositories_file':
+            warnings.warn(
+                "'colcon_mixin.mixin.repository.mixin_repositories_file' has "
+                "been deprecated, use 'colcon_mixin.mixin.repository."
+                "get_mixin_repositories_file()' instead", stacklevel=2)
```

Comment:
```suggestion
                "get_mixin_repositories_file()' instead",
                DeprecationWarning, stacklevel=2)
```
[PEP 387](https://peps.python.org/pep-0387/#making-incompatible-changes) suggests adding [`DeprecationWarning`](https://docs.python.org/3/library/exceptions.html#DeprecationWarning) as the warning category for usage.

In anticipation of this change, I have added suggestions to the tests to match this call.

Reply:
This package already uses `UserWarning` for deprecations [elsewhere](https://github.com/colcon/colcon-mixin/blob/5619db705a5db04ce5bd7b0d055851e53b509741/colcon_mixin/mixin/mixin_argument.py#L55). The main reason is that `DeprecationWarning` messages are not shown to on the console and are effectively silent unless specifically enabled: https://docs.python.org/3/library/warnings.html#default-warning-filter

Reply:
Thank you for the explanation, Scott. The remaining associated comments are related to this addition and can therefore be ignored.